source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -355,29 +397,85 @@ Before a connection can be returned outside of the pool, it must be “set up”
+
+If minPoolSize is set, the pool MUST ensure that it always has at
+least minPoolSize total `Connections <#connection>`_. In order to achieve this, it
+needs to be able to create `Connections <#connection>`_ that begin as "available"
+instead of "in use" (i.e. "populate the pool"). Performing this
+process MUST NOT block any application threads. For example, it could
+be performed on a background thread or via the use of non-blocking
+I/O.
+
+.. code::
+
+   create connection
+   establish connection
+   mark connection as available
+
 
 Checking Out a Connection
 -------------------------
 
-A Pool MUST have a method of allowing the driver to check out a Connection. Checking out a Connection involves entering the WaitQueue, and waiting for a Connection to become available. If the thread times out in the WaitQueue, an error is thrown.
+A Pool MUST have a method of allowing the driver to check out a `Connection <#connection>`_. Checking out a `Connection <#connection>`_ involves entering the WaitQueue and waiting for a `Connection <#connection>`_ to become available. If the thread times out in the WaitQueue, an error is thrown.
 
-If, in the process of iterating available connections in the pool by the checkOut method, a perished connection is encountered, such a connection MUST be closed and the iteration of available connections MUST continue until either a non-perished available connection is found or the list of available connections is exhausted. If no connections are available and the total number of connections is less than maxPoolSize, the pool MUST create and return a new Connection.
+If, in the process of iterating available `Connections <#connection>`_ in the pool
+by the checkOut method, a perished `Connection <#connection>`_ is encountered, such
+a `Connection <#connection>`_ MUST be closed and the iteration of available
+`Connections <#connection>`_ MUST continue until either a non-perished available
+`Connection <#connection>`_ is found or the list of available `Connections <#connection>`_ is
+exhausted. If no `Connections <#connection>`_ are available and the total number of
+`Connections <#connection>`_ is less than maxPoolSize, the pool MUST create and
+return a new `Connection <#connection>`_.
 
-If the pool is closed, any attempt to check out a connection MUST throw an Error, and any items in the waitQueue MUST be removed from the waitQueue and throw an Error.
+If the pool is closed, any attempt to check out a `Connection <#connection>`_ MUST throw an Error, and any items in the waitQueue MUST be removed from the waitQueue and throw an Error.
 
-If minPoolSize is set, the Connection Pool must have at least minPoolSize total connections. If the pool does not implement a background thread as specified here, the checkOut method is responsible for ensuring this requirement.
+If minPoolSize is set, the `Connection <#connection>`_ Pool MUST always have at
+least minPoolSize total `Connections <#connection>`_. If the pool does not
```

Comment:
I believe the requirement of the original CMAP spec was min pool size of *available + in use* connections per another earlier comment, and I also find this requirement impractical to implement.

I believe simply changing to "total" connections also produces a poorly specified connection pool.

I suggest either changing this paragraph to use available + in use connections as per the original intent of the spec, or perhaps removing this paragraph completely.